nideep/datasets/washington-rgbd/washington_rgbd.py

```diff
@@ -0,0 +1,270 @@
+import os
+import numpy as np
+import cv2
+import argparse
+import shutil
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+
+class WashingtonRGB:
+    ROOT_DEFAULT = '/mnt/raid/data/ni/dnn/pduy/rgbd-dataset'
+    CSV_DEFAULT = '/mnt/raid/data/ni/dnn/pduy/rgbd-dataset/rgbd-dataset.csv'
+    CSV_INTERPOLATED_DEFAULT = '/mnt/raid/data/ni/dnn/pduy/rgbd-dataset/rgbd-dataset-interpolated.csv'
+
+    def __init__(self, root_dir='', output=''):
+        self.root_dir = root_dir if root_dir != '' else self.ROOT_DEFAULT
+        self.output = output
+
+    # Load the dataset metadata to a Pandas dataframe and save the result to a csv file 
+    # if it did not exists
+    # The missing pose values will be saved as -1
+    def load_metadata(self, csv_output=''):
+        if os.path.exists(self.CSV_DEFAULT):
+            print 'reading from ' + self.CSV_DEFAULT
+            return pd.read_csv(self.CSV_DEFAULT)
+
+        file_list = os.walk(self.root_dir)
+
+        data = []
+
+        for current_root, dirs, files in file_list:
+            files = [f for f in files if 'mask' not in f and 'loc' not in f and 'pose' not in f]
+
+            for f in files:
+                pose_value = -1
+
+                name_components = f.split('_')
+                # reverse the list to void the cases when the category name has an "_"
+                reversed_name_components = np.flip(name_components, axis=0)
+
+                if len(reversed_name_components) < 5:
+                    continue
+
+                if (len(reversed_name_components) > 5):
+                    category = '_'.join(name_components[0: len(name_components) - 4])
+                else:
+                    category = reversed_name_components[4]
+
+                instance_number = reversed_name_components[3]
+                video_no = reversed_name_components[2]
+                frame_no = reversed_name_components[1]
+                data_type = reversed_name_components[0].split('.')[0]
+
+                # if int(frame_no) % 5 == 1:
+                name_components[len(name_components) - 1] = 'pose.txt'
+                pose_file_name = '_'.join(name_components)
+
+                try:
+                    with open(current_root + '/' + pose_file_name, 'r') as pose_file:
```

Comment:
use os.path.join(current_root, pose_file_name) instead of concatenating a file separator.